Thanks for the comments and the suggestion! please also delight with me in my closing/reopening the ticket -- proper newcomer :) I have a few questions:

1. I think the downstream C-function takes care of swapping o_stop for o_start if o_start is not provided, if I am reading it correctly? If so, I would then leave the block out. 

2. I also want to confirm then that is not an enhancement: that the goal is also to keep the current behaviour of not allowing `arange(stop=4)` currently:
```
> np.arange(3)
array([0, 1, 2])
> np.arange(stop=3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: arange() missing required argument 'start' (pos 1)
```
3. Confirming that the tests are in the appropriate spot? I wasn't able to find any arange tests so I'm not confident in where I put these. I'll hope that when I update the ^ code, the coverage issue will go away. 

re:
> Ah, looking at it again. The C-level function handles `stop == NULL` (meaning start is stop), but not not `start == NULL`, so the swapping is the other way around and it is a bit more complicated than I thought:
> 
> ```
> if o_stop == NULL:
>     # start is used as stop, but we reject this if it is passed by keyword
>     if args == NULL or len(args) == 0:  # could check `kwds != NULL` first 
>         # If keywords were used and no positional argument passed,
>         # start was passed by keyword and cannot be used as stop.
>         raise TypeError("uhoh, stop not passed!")
> else if o_start == NULL:
>     # Pass stop as NULL to indicate that start is stop
>     o_start = o_stop
>     o_stop = NULL    
> ```

--- a/numpy/core/src/multiarray/multiarraymodule.c
+++ b/numpy/core/src/multiarray/multiarraymodule.c
@@ -2930,7 +2930,7 @@ array_arange(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *kws) {
     static char *kwd[] = {"start", "stop", "step", "dtype", "like", NULL};
     PyArray_Descr *typecode = NULL;
 
-    if (!PyArg_ParseTupleAndKeywords(args, kws, "O|OOO&$O:arange", kwd,
+    if (!PyArg_ParseTupleAndKeywords(args, kws, "|OOOO&$O:arange", kwd,
                 &o_start,
                 &o_stop,
                 &o_step,
@@ -2938,6 +2938,18 @@ array_arange(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *kws) {
                 &like)) {
         Py_XDECREF(typecode);
         return NULL;
+    }
+
+    if (o_stop == NULL) {
+        if (args == NULL || PyTuple_GET_SIZE(args) == 0){
+            PyErr_SetString(PyExc_TypeError,
+                "arange() requires stop to be specified.");
+            return NULL;
+        }
+    }
+    else if (o_start == NULL) {
+        o_start = o_stop;
+        o_stop = NULL;
     }
 
     array_function_result = array_implement_c_array_function_creation(

--- a/numpy/core/tests/test_multiarray.py
+++ b/numpy/core/tests/test_multiarray.py
@@ -8295,6 +8295,22 @@ class TestArange:
         assert_raises(ZeroDivisionError, np.arange, 0, 0, 0)
         assert_raises(ZeroDivisionError, np.arange, 0.0, 0.0, 0.0)
 
+    def test_require_range(self):
+        assert_raises(TypeError, np.arange)
+        assert_raises(TypeError, np.arange, step=3)
+        assert_raises(TypeError, np.arange, dtype='int64')
+        assert_raises(TypeError, np.arange, start=4)
+
+    def test_start_stop_kwarg(self):
+        keyword_stop = np.arange(stop=3)
+        keyword_zerotostop = np.arange(start=0, stop=3)
+        keyword_start_stop = np.arange(start=3, stop=9)
+
+        assert len(keyword_stop) == 3
+        assert len(keyword_zerotostop) == 3
+        assert len(keyword_start_stop) == 6
+        assert_array_equal(keyword_stop, keyword_zerotostop)
+
 
 class TestArrayFinalize:
     """ Tests __array_finalize__ """

